File: file_handling/file_handling.go
package main

import (
	"fmt"
	"io/ioutil"
	// "os"
)

func main() {
	// !INFO: Writing to a file
	data := []byte("Hello, Go!")
	err := ioutil.WriteFile("output-go.txt", data, 0644)
	if err != nil {
		fmt.Println("Error:", err)
	} else {
		fmt.Println("File written successfully")
	}

	//!INFO: Reading from a file
	fileData, err := ioutil.ReadFile("output-go.txt")
	if err != nil {
		fmt.Println("Error:", err)
	} else {
		fmt.Println("File content:", string(fileData))
	}

	//!INFO: Deleting a file
	// err = os.Remove("output-go.txt")
	// if err != nil {
	// 	fmt.Println("Error:", err)
	// } else {
	// 	fmt.Println("File deleted successfully")
	// }
}

File: file_handling/file_handling.ts
import fs from "fs";

function main() {
  // Writing to a file
  fs.writeFile("output-ts.txt", "Hello, TypeScript!", "utf8", (err) => {
    if (err) {
      console.log("Error:", err);
    } else {
      console.log("File written successfully");
    }
  });

  // Reading from a file
  fs.readFile("output-ts.txt", "utf8", (err, data) => {
    if (err) {
      console.log("Error:", err);
    } else {
      console.log("File content:", data);
    }
  });

  // Deleting a file
  // fs.unlink("output-ts.txt", (err) => {
  //   if (err) {
  //     console.log("Error:", err);
  //   } else {
  //     console.log("File deleted successfully");
  //   }
  // });
}

main();
